### PR TITLE
fix duck not moving

### LINKED ISSUE #12333: Amusing Duck Does Not Move
### Labels

[BUG]

### Description

Title

### Steps to reproduce

1. Acquire amusing duck
2. It does not move
3. Cry

### Expected Behavior

For it to move

### Additional Information & Screenshots

Not my fault.

The `navigate_to` call in the duckbot move function always goes into a case used to detect movement to the same turf, even though it's not actually true. Might be an issue with `BOUNDS_DIST` or just some weird fluke with duck code specifically.

### PR DESCRIPTION
**Pull request by Laboredih123**
[BUG][MINOR][OBJECTS]

## About the PR 
make duck amusing by letting duck move
Fixes #12333

## Why's this needed? 
duck not amusing without movement

### CODE REVIEW

File: code/modules/robotics/bot/duckbot.dm [hunk 1]
```diff
@@ -87,6 +87,19 @@
 	src.botcard = new /obj/item/card/id(src)
 	src.botcard.access = get_access(src.access_lookup)
 
+/obj/machinery/bot/duckbot/navigate_to(atom/the_target, var/move_delay = 10, var/adjacent = 0, max_dist=60)
+	var/target_turf = get_pathable_turf(the_target)
+	if((get_dist(the_target, src) == 0))
```
Comment (stonepillars) [inline code/modules/robotics/bot/duckbot.dm:92]: What's `var/adjacent` for again? If it's a boolean for whether or not it's an adjacent movement, you could just check for it in the original proc.

**Comment by pali6** [comment]
Any clue why BOUNDS_DIST breaks it in the first place?

**Comment by Laboredih123** [comment]
I mentioned this in the issue, basically bounds_dist returns 0 when the bounding boxes touch but not overlap. Since adjacent turfs, which only the duck goes to, will always be touching, 0 will always be returned, and thus the duck is not amusing.

File: code/modules/robotics/bot/duckbot.dm (hunk 1 above)
Comment (Laboredih123) [reply code/modules/robotics/bot/duckbot.dm:92]: It appears to be so that a bot stands next to the target tile, not on it

**Comment by pali6** [comment]
Ah yeah forgot about that. I wish bounds dist distinguished between +0 and -0 somehow

**Comment by Laboredih123** [comment]
can i still have my spacebux

**Comment by Laboredih123** [comment]
can i still have my spacebux

**Comment by pali6** [comment]
idk how the spacebux thing is done, it's not official Goonstation thing

**Pull request closed without merging**